Chembombs 2

nullbear opened the pull request:
Fixes issue with https://github.com/tgstation/-tg-station/pull/16475, allowing it to be merged. Easier and faster to do it this way.

nullbear committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

nullbear committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

nullbear committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master'

nullbear committed:
Adds Chembombs

git pisses me off.

This way is easier than trying to fix the issue with
https://github.com/tgstation/-tg-station/pull/16475

nullbear committed:
Updates to /turf/space type to /turf/open/space

phil235 commented on code/game/machinery/syndicatebomb.dm:
```diff
@@ -292,6 +342,128 @@
+			return
+		var/fraction = 1/reactable.len
+		for(var/atom/A in reactable)
+			reagents.reaction(A, TOUCH, fraction)
```
is this part copypasted from grenade code?

nullbear replied:
the alternative is hacky, or to implement global chem_mixing procs and change grenades to use those instead. This is faster and more efficient than either.

phil235 commented on code/game/machinery/syndicatebomb.dm:
```diff
@@ -292,6 +342,128 @@
+	if(istype(I, /obj/item/weapon/crowbar) && beakers.len > 0)
+		for (var/obj/item/B in beakers)
+			B.loc = get_turf(src)
+			beakers -= B
```
no sound when crowbaring?

nullbear replied:
this is actually supposed to be "attack_self". forgot to change it.

nullbear committed:
Adds Crowbar sounds. Also Chem_Splash proc.

Adds crowbar sound to attackby proc.
Adds chem_splash proc, a global proc for reacting chemicals like a chem
grenade.
Made grenades and large grenades add a tiny bit of heat to their
reagents when exploding, making it easier to set off some reactions in
grenades.
Added "no_react" to trans_to(), making it possible to mix more than 2
beakers at once without it exploding in your face.
Made large grenades 10% larger.

nullbear committed:
Updates to changelog

nullbear committed:
Fixes grenades having 0 threat. Renames to threatscale.

nullbear commented:
better?

nullbear committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into chembombs-2

nullbear committed:
Adds more grenade types.

Adds 3 types of grenades:
Cryo: A grenade that cools its contents upon detonation. Has a smaller,
denser, blast radius.
Pyro: A grenade that ignites its contents upon detonation. Also heats
them a fair amount more.
Adv_Release: A grenade that will only detonate X amount of units, making
repeating time or prox sensors useful in grenades. They can be
configured with a multitool.

When building a chemical bomb payload, the type of grenade casings used
will affect the bombs properties.

nullbear committed:
Less Sparks

Less sparks produced. pyro bombs ignite mixture properly.

Adv. Release bombs take longer to finish detonating.

Changes payload sprite.

nullbear committed:
Merge remote-tracking branch 'refs/remotes/tgstation/master' into chembombs-2

# Conflicts:
#	code/modules/crafting/recipes.dm

nullbear committed:
Fixes merge conflict. Hopefully?

nullbear committed:
Adds grenades to rnd

Adds new grenade casings to RnD

nullbear commented:
@KorPhaeron Anything that needs to be done with this, to have it merged?

KorPhaeron commented:
Get someone who knows how to code to review it would be the most important part

nullbear commented:
@phil235 You're pretty good at calling out my laziness. You wanna review this?

GunHog commented:
Just run your final code locally, make sure everything works (I cannot use the "t" word), and fix anything that is broken.

GunHog commented on code/modules/research/designs/weapon_designs.dm:
```diff
@@ -127,6 +127,42 @@
+*/
+
+/datum/design/adv_grenade
+	name = "Advanced Release Grenade"
```
A....LIGHT GRENADE?

nullbear replied:
The cryo im thinking would be for stuff like freezing wet floors into ice, or frostoil stuff, etc. idk man. It's just intended as a compliment to the pyro.

I tested a flashbang 'advanced release bomb' with 4 bluespace beakers, and it's a pretty effective area denial device. If you enter the radius, you're flashed until you go blind.

GunHog commented on code/modules/research/designs/weapon_designs.dm:
```diff
@@ -127,6 +127,42 @@
 
+/datum/design/pyro_grenade
+	name = "Pyro Grenade"
+	desc = "An advanced grenade that is able to self ignite its mixture."
```
This thing will be **GOOFTASTIC**

GunHog commented on code/game/atoms.dm:
```diff
@@ -1,404 +1,404 @@
-			var/datum/reagent/consumable/nutri_check = R
-			if(nutri_check.nutriment_factor >0)
-				M.reagents.remove_reagent(R.id,R.volume)
+/atom
```
Try to find out what made Github think you removed and readded all these lines.

nullbear replied:
This was the merge update. So someone else did this.

nullbear replied:
Yeah, looking at it. What happened is i did an update from master. And didn't look through it really. I'd have had to look through it pretty meticulously to notice this, and then discard the changes. Nothing is changed though, so it wont be an issue.

Also accidently didn't include a couple files as part of the update from master. Included those now.

nullbear committed:
update from tgstation master

had dmaker open, so new files werent automatically included. whoopsies!

nullbear commented:
atoms.dm file is just github being dumb. Won't have any impact, since nothing is changed.

nullbear commented:
okay i think i figured it out. i'm not used to using the shell.
